### ISSUE TITLE
Impossible to configure referenced endity field differently per content type

### ISSUE THREAD

**Issue by alanmels** [backdrop/backdrop-issues#6706]
## Description of the bug

I needed to reference the same field in two different content fields, but configure them differently. While one of two content types using the referenced field needs only 1 limited number of values, the second one needs to have unlimited number of values. However, as soon as I change the number of values for the references field in one of the content types settings, it also changes the same setting option for the other content type.

## Steps To Reproduce

To reproduce the behavior:

1. Reference a field from a content type and set the `Number of values` to limited and save. 
2. Open another window and reference the same field from a second content type and set the `Number of values` to unlimited and save.
3. Clear cache. 
4. Reload the first window and see the `Number of values` is set to unlimited.

## Additional information

- Maybe we need to open a separate bug report for step 3 as cache should have been cleared when a content type is saved.
- This is not related, but I wonder if such contributed modules as https://github.com/backdrop-contrib/references and https://github.com/backdrop-contrib/reference should be retired as since entity reference was added to core. At least their README's must be updates as the second one has quite confident text which could confuse some newer users:

> This module is the sucessor to both references and entity_reference in Drupal 7, and will one day include all the features from both, along with an upgrade path. When that has been accomplished, this module will be moved into Backdrop core.

**Comment by stpaultim** [comment]
https://github.com/backdrop-contrib/reference/issues/75

**Comment by argiepiano** [comment]
Correct. You can't have the same field with different cardinality. This is a Global setting of the field. 

But what you could do is use a form alter hook to modify the actual form element, in the form itself, so that the user can't click an Add more button.

**Comment by alanmels** [comment]
@olafgrabienski, yes, it appears some field settings are global, and this fact itself can be confusing to users while adjusting field settings on content type level. Ideally, there should be interface page for fields to configure global settings with the content type level to configure only those settings that are possible to set only for that particular content type. So if this is not a bug, then I'd re-classify this as a feature request. The current state is really confusing.

Using two different fields is not an option for the purpose of the project we are working on (the same field is used in multiple places - custom modules, views pages, templates, etc.), so for now we will go with the option suggested by @argiepiano.

**Comment by olafgrabienski** [comment]
@alanmels I agree, the current behavior is not very clear. There are already some related issue reports with suggestions to mitigate the confusion:

- https://github.com/backdrop/backdrop-issues/issues/3349
- https://github.com/backdrop/backdrop-issues/issues/3362
- https://github.com/backdrop/backdrop-issues/issues/5345

Do you have other ideas how to improve the field creation process?
